server/registry/names/api.go

```diff
@@ -28,12 +28,16 @@ type Api struct {
 // Validate returns an error if the resource name is invalid.
 // For backward compatibility, names should only be validated at creation time.
 func (a Api) Validate() error {
+	if err := validateID(a.ApiID); err != nil {
+		return err
+	}
```

Comment:
I'm surprised that this is untested because we have [this test case](https://github.com/apigee/registry/blob/616e3a92c71fbb528430f649700215f8ec3c9f06/server/registry/names/names_test.go#L88). Did we miss the real source of the error message? It seems that to get here we need a `names.Api`, and we might be getting that with [names.ParseApi()](https://github.com/apigee/registry/blob/616e3a92c71fbb528430f649700215f8ec3c9f06/server/registry/names/api.go#L98), which will return the error that we don't want to see before we get here.

Reply:
I tested this manually and got the errors I put in the description, so... no?